projects/fal/src/fal/api.py

```diff
@@ -503,6 +512,15 @@ def register(
             request_timeout=request_timeout,
             startup_timeout=startup_timeout,
         )
+        files = options.host.get("files", [])
+        if files:
+            sync = FileSync(self.local_file_path)
+            result = asyncio.run(sync.sync_files(files))
```

Comment:
you should never use asyncio.run inside a sync function that is not top level. either make this function async, or make sync_files work syncronusly.